test/src/userobjects/InterfaceDomainUserObject.C

```diff
@@ -0,0 +1,164 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceDomainUserObject.h"
+#include "MooseMesh.h"
+#include "Function.h"
+#include "MooseVariableFE.h"
+#include "Assembly.h"
+
+registerMooseObject("MooseTestApp", InterfaceDomainUserObject);
+
+InputParameters
+InterfaceDomainUserObject::validParams()
+{
+  InputParameters params = DomainUserObject::validParams();
+  params.addRequiredCoupledVar("u", "The variable we are diffusion for");
+  params.addRequiredCoupledVar("v", "The variable on the other side of the interface");
+  params.addParam<FunctionName>("function", "1", "The forcing function.");
+  params.addParam<Real>("robin_coef", 2, "The cofficient multiplying the Robin boundary condition");
+  params.addParam<std::vector<BoundaryName>>(
+      "robin_boundaries", "The boundaries on which to apply the Robin boundary condition");
+  params.addParam<std::vector<BoundaryName>>(
+      "interface_boundaries",
+      "The interface boundaries on which to apply the continuity of flux and value conditions");
+  params.addRequiredParam<Real>(
+      "interface_penalty",
+      "The penalty that penalizes jump between primary and neighbor variables.");
+  params.addRequiredParam<Real>("nl_abs_tol", "The absolute tolerance of the nonlinear solver");
+  return params;
+}
+
+InterfaceDomainUserObject::InterfaceDomainUserObject(const InputParameters & parameters)
+  : DomainUserObject(parameters),
+    _u(coupledValue("u")),
+    _v_neighbor(coupledNeighborValue("v")),
+    _grad_u(coupledGradient("u")),
+    _grad_v_neighbor(*coupledNeighborGradients("v")[0]),
+    _var(*getVar("u", 0)),
+    _v_var(*getVar("v", 0)),
+    _test(_var.phi()),
+    _grad_test(_var.gradPhi()),
+    _test_face(_var.phiFace()),
+    _grad_test_face(_var.gradPhiFace()),
+    _test_face_neighbor(_var.phiFaceNeighbor()),
+    _grad_test_face_neighbor(_var.gradPhiFaceNeighbor()),
+    _func(getFunction("function")),
+    _robin_coef(getParam<Real>("robin_coef")),
+    _interface_penalty(getParam<Real>("interface_penalty")),
+    _nl_abs_tol(getParam<Real>("nl_abs_tol"))
+{
+  const auto & robin_boundaries = getParam<std::vector<BoundaryName>>("robin_boundaries");
+  const auto & robin_bnd_ids_vec = _mesh.getBoundaryIDs(robin_boundaries);
+  _robin_bnd_ids = std::set<BoundaryID>(robin_bnd_ids_vec.begin(), robin_bnd_ids_vec.end());
+
+  const auto & interface_boundaries = getParam<std::vector<BoundaryName>>("interface_boundaries");
```

Comment:
Either give default empty values in the parameter or have isParamValid check around this.